packages/plugin-react-oxc/src/index.ts

```diff
@@ -0,0 +1,158 @@
+import { dirname, join } from 'node:path'
+import { fileURLToPath } from 'node:url'
+import { readFileSync } from 'node:fs'
+import type { BuildOptions, Plugin, PluginOption } from 'vite'
+import {
+  addRefreshWrapper,
+  avoidSourceMapOption,
+  getPreambleCode,
+  runtimePublicPath,
+  silenceUseClientWarning,
+} from '@vitejs/react-common'
+
+const _dirname = dirname(fileURLToPath(import.meta.url))
+
+export interface Options {
+  include?: string | RegExp | Array<string | RegExp>
+  exclude?: string | RegExp | Array<string | RegExp>
+  /**
+   * Control where the JSX factory is imported from.
+   * @default 'react'
+   */
+  jsxImportSource?: string
+}
+
+const defaultIncludeRE = /\.[tj]sx?(?:$|\?)/
+
+export default function viteReact(opts: Options = {}): PluginOption[] {
+  const include = opts.include ?? defaultIncludeRE
+  const exclude = [
+    ...(Array.isArray(opts.exclude)
+      ? opts.exclude
+      : opts.exclude
+        ? [opts.exclude]
+        : []),
+    /\/node_modules\//,
+  ]
+
+  const jsxImportSource = opts.jsxImportSource ?? 'react'
+  const jsxImportRuntime = `${jsxImportSource}/jsx-runtime`
+  const jsxImportDevRuntime = `${jsxImportSource}/jsx-dev-runtime`
+
+  const viteConfig: Plugin = {
+    name: 'vite:react-oxc:config',
+    config(userConfig, { command }) {
+      return {
+        // @ts-expect-error rolldown-vite Vite type incompatibility
+        build: silenceUseClientWarning(userConfig) as BuildOptions,
+        oxc: {
+          jsx: {
+            runtime: 'automatic',
+            importSource: jsxImportSource,
+            refresh: command === 'serve',
+            development: command === 'serve',
+          },
+          jsxRefreshInclude: include,
+          jsxRefreshExclude: exclude,
+        },
+        optimizeDeps: {
+          include: [
+            'react',
+            'react-dom',
+            jsxImportDevRuntime,
+            jsxImportRuntime,
+          ],
+          rollupOptions: { jsx: { mode: 'automatic' } },
+        },
+      }
+    },
+    options() {
+      if (!this.meta.rolldownVersion) {
+        throw new Error(
+          '@vitejs/plugin-react-oxc requires rolldown-vite to be used. ' +
+            'See https://vitejs.dev/guide/rolldown for more details about rolldown-vite.',
+        )
+      }
+    },
+  }
+
+  const viteRefreshRuntime: Plugin = {
+    name: 'vite:react-oxc:refresh-runtime',
+    enforce: 'pre',
+    resolveId: {
+      filter: { id: exactRegex(runtimePublicPath) },
+      handler(id) {
+        return id
+      },
+    },
+    load: {
+      filter: { id: exactRegex(runtimePublicPath) },
+      handler(_id) {
+        return readFileSync(
+          join(_dirname, 'refresh-runtime.js'),
```

Comment:
Note related to this PR because the issue also exist in the babel plugin but this is not working in dev.
I not sure yet what is the best way to handle this (I though about adding `src/refresh-runtime.js` to gitignore and having a copy step in the dev script, but not 100% convinced)

Reply:
I'll take a look if I can fix this in a separate PR 👍